src/SignalR/server/Core/src/HubMetadata.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public class HubMetadata
+    {
+        public Type HubType { get; set; }
```

Comment:
Do you really want this to be mutable? Everythign we put in metadata is immutable so far.

Reply:
Ah good catch.

Reply:
Bleh, making it immutable means constructor hell though.

Reply:
oh no, a single constructor with a single parameter :trollface:

Reply:
It's fine for this since HubType is *required* anyways. It's the fact that other potentially optional metadata will need to be get only and will require more ctors